### PR TITLE
Make `riff` and `riff-distro` use consistent manifest declarations

### PR DESCRIPTION
**Pull request by glyn**
It is important that the manifests are consistent so that distributions built with `riff-distro create` can refer to the same stable and latest values as `riff`.

This change ensures that manifests are only declared in main packages (to avoid the possibility of direct imports bypassing dependency injection), but shares the declarations using a symbolically linked file to avoid dual maintenance.

/cc @scothis @trisberg

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,7 +37,7 @@ install: build
 	cp $(OUTPUT) $(GOBIN)
 
 $(OUTPUT): $(GO_SOURCES) vendor VERSION
-	go build -o $(OUTPUT) -ldflags "$(LDFLAGS_VERSION)" main.go
+	go build -o $(OUTPUT) -ldflags "$(LDFLAGS_VERSION)" *.go
```
Comment (ericbottard) [inline Makefile]: Why is this necessary?

Comment (glyn) [reply Makefile]: To ensure that `main.go` and `configuration.go` are compiled together. When it specifies `main.go`, the code does not compile.

Comment (scothis) [reply Makefile]: we should explicitly specify both main.go and configuration.go in order to avoid future confusion

`make release` also needs to be updated.

Comment (ericbottard) [reply Makefile]: I'm not sure I understand *why* this is necessary, I thought `go build` found all required files (just the same way it knows how to find referenced packages).

Anyway, we could also change from "build these `.go` files" to "build this package"

Comment (georgeharley) [reply Makefile]: FWIW @ericbottard's suggestion of just building the package by not specifying any go source files at all in the `go build` invocations works fine for both the `make build` and `make release` targets.

**Comment by glyn** [comment]
Merged as 8cc21f1d6a84d973cd62aa4ade84c0538f88eddb
